File: old/tooko/td/core/LongLongArrayMap.java
package tooko.td.core;

import java.util.*;

public class LongLongArrayMap extends HashMap<Long, LinkedList<Long>> {

    public LinkedList<Long> rawGet(Object key) {

        return super.get(key);

    }

    @Override
    public LinkedList<Long> get(Object key) {

        LinkedList<Long> list = super.get(key);

        if (list == null) {

            list = new LinkedList<>();

            put((Long) key, list);

        }

        return list;

    }

    public int countElements() {

        int count = 0;

        for (LinkedList<Long> elements : values()) {

            count += elements.size();

        }

        return count;

    }

}

File: src/main/java/tookox/pm/LongLongArrayMap.kt
package tookox.pm

import java.util.*

class LongLongArrayMap : HashMap<Long, LinkedList<Long>>() {

    fun rawGet(key: Long): LinkedList<Long>? = super.get(key)

    override fun get(key: Long): LinkedList<Long> {

        var list = super.get(key)

        if (list == null) {

            list = LinkedList()

            put(key, list)

        }

        return list

    }

    fun countElements(): Int {

        var count = 0

        for (elements in values) {

            count += elements.size

        }

        return count
    }

}